TigerRag/demos/langchain/demo.py

```diff
@@ -98,20 +117,40 @@ def main():
     for index in range(min(num_of_run, len(data["queries"]))):
         question = data["queries"][index]
         # Example usage of EBR
+        loader = WebBaseLoader(documents)
+        text_splitter = RecursiveCharacterTextSplitter(
+            chunk_size=500, chunk_overlap=0)
+        splits = text_splitter.split_documents(loader.load())
+
         vectorstore = get_documents_embeddings(documents)
         print("The following is EBR output for question: "+question)
         retrieved_context = ebr(question, vectorstore)
         print(retrieved_context)
 
         # Example usage of RAG
         print("The following is RAG output for question: "+question)
-        print(generate_answer_with_rag_gpt3(
-            question, retrieved_context, 'text-davinci-003'))
+        # print(generate_answer_with_rag_gpt3(
+        #     question, retrieved_context, 'text-davinci-003'))
+
+        # Retrivel Augmented Generation
+        prompt_rag = f"Context: {retrieved_context} Question: {question}. Provide a summary or answer:"
+
+        openai_generative_retrieval_augmenter = OpenAIRetrievalAugmenter(
+            "text-davinci-003")
+        answer_rag = openai_generative_retrieval_augmenter.get_augmented_retrieval(
+            prompt_rag)
+        print(answer_rag)
 
         # Example usage of GAR
         print("The following is GAR output for question: "+question)
-        print(generate_answer_with_gar_gpt3(
-            question, retrieved_context, 'text-davinci-003', vectorstore))
+        # print(generate_answer_with_gar_gpt3(
+        #     question, retrieved_context, 'text-davinci-003', vectorstore))
+        prompt_gar = prompt = f"Expand on the query: {question}"
+        openai_generative_query_augmenter = OpenAIQueryAugmenter(
+            "text-davinci-003")
```

Comment:
This seems to be duplicate of the code you updated in generate_answer_with_gar_gpt3()